common/cluster/metadata.go

```diff
@@ -450,7 +459,8 @@ func (m *metadataImpl) updateFailoverVersionToClusterName() {
 func updateVersionToClusterName(clusterInfo map[string]ClusterInformation, failoverVersionIncrement int64) map[int64]string {
 	versionToClusterName := make(map[int64]string)
 	for clusterName, info := range clusterInfo {
-		if failoverVersionIncrement <= info.InitialFailoverVersion || info.InitialFailoverVersion <= 0 {
+		if clusterName != FakeClusterForEmptyVersion &&
+			(failoverVersionIncrement <= info.InitialFailoverVersion || info.InitialFailoverVersion <= 0) {
```

Comment:
how about keeping the existing logic and add the `copyClusterInfo[FakeClusterForEmptyVersion] = fakeClusterInfo` to the very end?

Reply:
e.g. validate as is, then add the additional FakeClusterForEmptyVersion afterwards

Reply:
The versionToClusterName uses in ClusterNameForFailoverVersion(), which already been handled. You can just ignore the fake cluster